wavenet/audio_reader.py

```diff
@@ -94,18 +96,28 @@ def thread_main(self, sess):
                               "silence. Consider decreasing trim_silence "
                               "threshold, or adjust volume of the audio."
                               .format(filename))
-
-                if self.sample_size:
-                    # Cut samples into fixed size pieces
-                    buffer_ = np.append(buffer_, audio)
-                    while len(buffer_) > self.sample_size:
-                        piece = np.reshape(buffer_[:self.sample_size], [-1, 1])
+                    if audio.size <= self.receptive_field_size:
+                        print("Warning: {} was ignored since, after silence "
+                              "trimming, it was shorter than the receptive "
+                              "field of the net. ".format(filename))
+
+                # If the trimmed audio is less than the receptive field size,
+                # then tf.slice performed by loss will fail. So skip the file
+                # if this is the case.
+                if audio.size > self.receptive_field_size:
+                    if self.sample_size:
```

Comment:
I know this is not your code, but since we're at it -- sample_size is not a boolean, so we should explicitly test for `is not None`, or `> 0`, or whatever appropriate.